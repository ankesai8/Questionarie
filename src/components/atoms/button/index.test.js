import React from "react";
import renderer from "react-test-renderer";
import ShallowRenderer from "react-test-renderer/shallow";

import { Button } from "./index";

const getChildrenByClass = (tree, classString) => {
  const result = tree.root.find(
    (instance) => instance.props.className === classString
  ).props.children[0];
  return result;
};

describe("Button", () => {

  it("Button should render properly", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<Button type="submit" text={"Submit"} />);
    expect(tree).toMatchSnapshot();
  });
});

describe("Button  UI", () => {

  const getButtonType = (tree, classString) => {
    const result = tree.root.find(
      (instance) => instance.props.className === classString
    ).props.type;
    return result;
  };

  it("should show the button text", () => {
    const tree = renderer.create(<Button type="submit" text={"Submit"} />);
    const text = getChildrenByClass(tree, "button");

    expect(text).toBe("Submit");

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it("should show the button type", () => {
    const tree = renderer.create(<Button type="submit" text={"Submit"} />);
    const type = getButtonType(tree, "button");

    expect(type).toBe("submit");

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
