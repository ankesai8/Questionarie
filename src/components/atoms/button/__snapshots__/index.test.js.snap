// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button  UI should show the button text 1`] = `
<button
  className="button"
  type="submit"
>
  Submit
  <span
    className="icon"
  >
    →
  </span>
</button>
`;

exports[`Button  UI should show the button type 1`] = `
<button
  className="button"
  type="submit"
>
  Submit
  <span
    className="icon"
  >
    →
  </span>
</button>
`;

exports[`Button Button should render properly 1`] = `
<button
  className="button"
  type="submit"
>
  Submit
  <span
    className="icon"
  >
    →
  </span>
</button>
`;
